src/util/segmentation/operations/correction.js

```diff
@@ -0,0 +1,524 @@
+
+  for (let i = 0; i < pixelPath.length - 1; i++) {
+    const { left, right } = getNodesPerpendicularToPathPixel(
+      pixelPath[i],
+      pixelPath[i + 1]
+    );
+
+    leftPath.push(left);
+    rightPath.push(right);
+  }
+
+  return { pixelPath, leftPath, rightPath };
+}
+
+/**
+ * GetNodesPerpendicularToPathPixel - Using the current and next pixel on the path, determine the adjacent pixels
+ * which are perpendicular to the path direction. (i.e. to the left and to the right).
+ *
+ * @param  {Object} pathPixel The pixel being queried.
+ * @param  {Object} nextPathPixel the pathPixel's successor.
+ *
+ * @returns {Object} The coordinates of the left and right perpendicular pixels.
+ */
+function getNodesPerpendicularToPathPixel(pathPixel, nextPathPixel) {
+  const direction = {
+    x: nextPathPixel.x - pathPixel.x,
+    y: nextPathPixel.y - pathPixel.y,
+  };
+
+  // P = pathPixel, n = nextPathPixel, L = left, R = right
+  //
+  // |n| |R|  | |n| |  |L| |n|
+  // | |p| |  |L|p|R|  | |p| |
+  // |L| | |  | | | |  | | |R|
+  //
+  // | |R| |           | |L| |
+  // |n|p| |           | |p|n|
+  // | |L| |           | |R| |
+  //
+  // |R| | |  | | | |  | | |L|
+  // | |p| |  |R|p|L|  | |p| |
+  // |n| |L|  | |n| |  |R| |n|
+
+  if (direction.x === -1 && direction.y === 1) {
+    return {
+      left: { x: pathPixel.x - 1, y: pathPixel.y - 1 },
+      right: { x: pathPixel.x + 1, y: pathPixel.y + 1 },
+    };
+  } else if (direction.x === 0 && direction.y === 1) {
+    return {
+      left: { x: pathPixel.x - 1, y: pathPixel.y },
+      right: { x: pathPixel.x + 1, y: pathPixel.y },
+    };
+  } else if (direction.x === 1 && direction.y === 1) {
+    return {
+      left: { x: pathPixel.x - 1, y: pathPixel.y + 1 },
+      right: { x: pathPixel.x + 1, y: pathPixel.y - 1 },
+    };
+  } else if (direction.x === 1 && direction.y === 0) {
+    return {
+      left: { x: pathPixel.x, y: pathPixel.y + 1 },
+      right: { x: pathPixel.x, y: pathPixel.y - 1 },
+    };
+  } else if (direction.x === 1 && direction.y === -1) {
+    return {
+      left: { x: pathPixel.x + 1, y: pathPixel.y + 1 },
+      right: { x: pathPixel.x - 1, y: pathPixel.y - 1 },
+    };
+  } else if (direction.x === 0 && direction.y === -1) {
+    return {
+      left: { x: pathPixel.x + 1, y: pathPixel.y },
+      right: { x: pathPixel.x - 1, y: pathPixel.y },
+    };
+  } else if (direction.x === -1 && direction.y === -1) {
+    return {
+      left: { x: pathPixel.x + 1, y: pathPixel.y - 1 },
+      right: { x: pathPixel.x - 1, y: pathPixel.y + 1 },
+    };
+  } else if (direction.x === -1 && direction.y === 0) {
+    return {
+      left: { x: pathPixel.x, y: pathPixel.y - 1 },
+      right: { x: pathPixel.x, y: pathPixel.y + 1 },
+    };
+  }
+
+  logger.error(
+    `Unable to find left and right paths for flood fill `,
+    pathPixel,
+    nextPathPixel,
+    direction
+  );
+}
+
+/**
+ * SplitLineIntoSeperateOperations - Splits the path of nodes into
```

Comment:
You don't need to prefix methods with the name here. It'll show up automatically w/ intellisense as the function is a named function. If you really want to specify it, you can use the `@function`, `@method`, or `@name` JSDoc options?

Reply:
Na I agree, if we don't need to add it, lets add the minimal that's needed. Can we setup prettier/eslint to maybe add a doc template on save thats a consistent format? If that's possible that might be a good idea.

Reply:
Does VS Code note scaffold a JS Doc comment for you when you type `/**` and hit enter?

Reply:
Most plugins just add `params`, I think we always want to specify a `returns` also, but prettier doesn't seem to add these for us currently?